### ISSUE TITLE
Cache dependency tree

### ISSUE THREAD

**Issue by russellpierce** [robertzk/lockbox#114]
Just walking the dependency tree for a lockfile can take a while.  This makes starting up with lockbox a tad slow.  It seems like one could digest the lockfile and store a cache of the dependency tree for subsequent start-ups to allow things to go a bit faster.

**Comment by robertzk** [comment]
Agreed, this should be specifiable by an option. In our use case, we update dependencies of dependencies frequently and typically the full walk is preferred on every startup, but I understand this is atypical for most environments.

**Comment by russellpierce** [comment]
For a given .yml the dependency tree may change:

1. If `autoinstall: true`
1. When package A depends on B and A is in the yml but B is not and B is updated between runs?

... Are there some other cases that immediately occur to you?
